### DISCUSSION TITLE
what about the natural numbers?

### DISCUSSION THREAD

**Discussion by zx9w** [coalton-lang/coalton#388] [Ideas and Proposals]
I don't mean to be snarky: https://www.youtube.com/watch?v=jFk1qpr1ytk

Fwict the stdlib doesn't have naturals so it is inheriting the stupidity of haskell. I'd be willing to contribute this if someone would give me some scaffold. If I'm wrong I'd be happy to hear it. Thanks for this wonderful project, do you have an irc channel?

**Comment by stylewarning** [comment]
There's a Discord linked in the README. 

If you have concrete suggestions for an interface and how it relates to the standard library, I'm all ears. We have some unsigned numbers. We could make an unsigned integer pretty easily, but you won't get "nice" indicative pattern matching.

If the natural number 1,000,000 is going to be represented as a nested structure proportional to the number's distance from zero, I'm not convinced it's useful in problems computers are good at solving. I'm also not convinced Haskell's choice of number design was "stupidity."

What sort of information do you need to get started? How can I help?

**Reply by zx9w** [reply]
Okay clearly you did not watch the linked video, but I won't blame you for that, it's a big ask.

Sorry for using adversarial phrasing I forget to switch vocabulary when I am on the internet and have much less body language.

> What sort of information do you need to get started? How can I help?

I guess not much at all, I just suffer a lot when I try to do things delibirately and I have a lot of fun when I do them accidentally. I was hoping that the latter may happen somehow by starting this discussion.

**Reply by stylewarning** [reply]
> Okay clearly you did not watch the linked video, but I won't blame you for that, it's a big ask.

I did watch it and it was entirely unconvincing.
